lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
+      fail HawkularException.new('Invalid credentials', http_code) if f.instance_of? RestClient::Unauthorized
```

Comment:
Shouldn't there be a message as part of the response that can be propagated? So there won't be a need to "guess" it or hardcode it